dpnp/backend/kernels/dpnp_krnl_arraycreation.cpp

```diff
@@ -110,15 +110,35 @@ void dpnp_tril_c(void* array_in,
                  const size_t ndim,
                  const size_t res_ndim)
 {
+    if ((array_in == nullptr) || (result1 == nullptr))
+    {
+        return;
+    }
+
     _DataType* array_m = reinterpret_cast<_DataType*>(array_in);
     _DataType* result = reinterpret_cast<_DataType*>(result1);
 
+    if ((shape == nullptr) || (res_shape == nullptr))
+    {
+        return;
+    }
+
+    if ((ndim == 0) || (res_ndim == 0))
+    {
+        return;
+    }
+
     size_t res_size = 1;
     for (size_t i = 0; i < res_ndim; ++i)
     {
         res_size *= res_shape[i];
     }
 
+    if (res_size == 0)
+    {
+        return;
+    }
```

Comment:
this looks like a some solution for  https://github.com/IntelPython/dpnp/pull/627/files#r583932215

But is it correct in terms of algorithm? Is this the behavior of `numpy` for such a given shape?
Do you have any tests for this case?

Reply:
@samir-nasibli  numpy return empty array in this case. For example:

>>> a=np.ndarray((5, 0, 3), dtype=np.int64)
>>> p=np.triu(a)
>>> p
array([], shape=(5, 0, 3), dtype=int64)

Reply:
Nice. Add please tests, that checks the same behaviour for noted case.